bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/MapValue.java

```diff
@@ -299,6 +301,24 @@ public BType getType() {
         return type;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void attemptFreeze(Status freezeStatus) {
+        if (this.type.getTag() == TypeTags.OBJECT_TYPE_TAG) {
```

Comment:
We have separate classes for objects now. Therefore we don't need this check.

Reply:
Fixed